### PR TITLE
Adds support to `wp-env` for using a `docker-compos.override.yml` file

### LINKED ISSUE #25486: Provide the ability for  `wp-env` to support a `docker-compose.override.yml` file for plugins that need additional Docker environment customization.
**Is your feature request related to a problem? Please describe.**
There is a need to support additional Docker local development customization for plugins when using the `wp-env` package. In the previous setup of the `env` command that was part of the `wp-scripts` package, a Docker Compose template could be provided that would be used in creating a `docker-compose.override.yml` file. My specific examples of this need are to provide additional Docker containers for integration testing against mock services. Another use case is trying to capture emails that WordPress sites send using a Mailhog container.

**Describe the solution you'd like**
The `wp-env` package should support the ability for plugins to include a `docker-compose.override.yml` file in their root directory. That file will be detected and used as the source for a new configuration variable `dockerComposeOverridePath`, to be included in the Docker startup. A default `docker-compose.override.yml` file could be provided/generated so that the NodeJS call to docker doesn't fail, or require logic to optionally include using an override file.

**Describe alternatives you've considered**
When advanced docker environment customization is required it seem infeasible to attempt to provide an extensive list of configuration properties that man not be using the majority of the time. Additionally, a large list of configuration properties to attempt to cover all use cases for all plugins seems unsustainable. I was considering providing a `.wp-env.config` property that could be set by the plugin/theme and/or override by a local `.wp-env.override.config` but I decided to keep this override capability more limited in scope and simplify the implementation.

### PR DESCRIPTION
**Pull request by timnolte**
## Description

This fixes #25486 by providing an option to include a `docker-compose.override.yml` file in the project directory that will be used as the source for a `docker-compose.override.yml` file in the working directory.

## How has this been tested?

Tested by starting up the local WordPress Gutenberg development environment and:
- Validated successful startup when not configured
- Validated services specified in the generated `docker-compose.override.yml` file, from the configured source file, have started up.

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/env/lib/commands/stop.js [hunk 1]
```diff
@@ -21,10 +21,15 @@ module.exports = async function stop( { spinner, debug } ) {
 		debug,
 	} );
 
+	const { dockerComposeOverrideConfigPath } = await initConfig( {
```
Comment (noahtallen) [inline packages/env/lib/commands/stop.js:24]: this should be moved to the existing call to `initConfig`, I think

**Review by noahtallen — CHANGES_REQUESTED** [review]
Sorry for letting this sit for so long!

I would think we only need to copy the override file to the working directory for docker-compose to automatically detect it. But since we are passing the override path to each of the docker-compose commands manually, it seems like we could just set that path directly to the file in the project directory. That way we don't have to copy the file over.

If we do need to copy the file over to add functionality, I don't think we should add any extra yaml packages for it. I think we could just use `fs.copyFile` or something like that since we don't need to use the file in code.

**Comment by moon0326** [comment]
Hi, ✋ I'm very interested in this PR. This PR will let us customize the docker build. Any chance this gets updated? Looks like this has been sitting here since last October. Thank you!

**Comment by timnolte** [comment]
The draft PR was opened then but I believe I have some more work done on this, but. It all of it. I had a lot of other work for 2 of my plugins that took me off of working in finalizing that PR. I'll try to pick it up again soon.

**Pull request closed without merging**